code/code/misc/gaining.cc

```diff
@@ -868,6 +895,114 @@ void TPerson::setSelectToggles(TBeing* gm, classIndT Class,
           "inventory. Where did that come from?",
           false, this, nullptr, nullptr, TO_CHAR, ANSI_CYAN);
       }
+
+      if (getLevel(Class) >= 15 && !hasQuestBit(TOG_THIEF_L15_ELIGIBLE) &&
+          !hasQuestBit(TOG_THIEF_L15_ACCEPTED) &&
+          !hasQuestBit(TOG_THIEF_L15_GAVE_PIPE) &&
+          !hasQuestBit(TOG_THIEF_L15_GAVE_MSG) &&
+          !hasQuestBit(TOG_THIEF_L15_GAVE_BOX) &&
+          !hasQuestBit(TOG_THIEF_L15_HAS_CARD)) {
+        setQuestBit(TOG_THIEF_L15_ELIGIBLE);
+
+        TNote* note = createNote(
+          format("%s -\n\n\rGuildmaster wants to see you. South marketplace in "
+                 "Grimhaven, behind Jennica's. Don't be followed.\n\n\r") %
+          getName());
+
+        *this += *note;
+
+        act(
+          "You suddenly realize there's an unfamiliar note in your "
+          "inventory. Where did that come from?",
+          false, this, nullptr, nullptr, TO_CHAR, ANSI_CYAN);
+      }
+
+      if (getLevel(Class) >= 20 && !hasQuestBit(TOG_THIEF_L15_ELIGIBLE) &&
```

Comment:
I'll probably just merge this but it seems like it's not an ideal way to handle this. We want to know two things: Are they eligible and have we already given them the message or note. However because we are untoggling stuff as we progress in the quest these statements look unwieldy. 

We could have a specific toggle TOG_GIVEN_THIEF_L20_NOTE or something that _doesn't_ get toggled off as the quest progresses and make this statement a lot easier.

Is that better? Or just better here but worse in some other way?